### DISCUSSION TITLE
VIPS in visual studio

### DISCUSSION THREAD

**Discussion by zjian361** [libvips/libvips#3118] [Help wanted]
I need to use libvips in visual studio. And I configured "include"、"lib" path and have tried much way to solve it ，but still can not run normally.The error always is unable to resolve external symbols on the VImage class' Members or others.
Could you give me a suggestion or a correct way to use it.Thanks very much!

**Comment by jcupitt** [comment]
Hi @zjian361,

C++ doesn't define an ABI, so you can't mix C++ compilers in a project. You must use exactly the same compiler (and often the same version of the same compiler) for every component you want to link.

We build the libvips windows binary using gcc, so you need to use g++ as your compiler on windows if you want to use `libvips-cpp.dll`, our C++ binding.

To use MSVC instead, you'll need to rebuild the libvips C++ binding. Just copy these files into your project:

https://github.com/libvips/libvips/tree/master/cplusplus

And link against `libvips-42.dll` instead.

**Reply by zjian361** [reply]
hi @jcupitt 
I try to read the mrxs image.But I find "VipsImage *in" size is not expected， it is so small size.I want to read the image that ‘’level = 0‘’.
![image](https://user-images.githubusercontent.com/116711055/200715435-201bf493-67d7-4864-8e4c-ed3e599c5565.png)
![image](https://user-images.githubusercontent.com/116711055/200717014-531abb0a-667b-4030-b993-9331e83e10c6.png)
VipsImage *in= vips_image_new_from_file("./20221008-BG27BN06F1-B3-KO14-2-GE-40X.mrxs", NULL);
Is the method of use incorrect？

**Reply by jcupitt** [reply]
I would guess you've not downloaded the libvips binary with openslide support. You need:

https://github.com/libvips/build-win64-mxe/releases/tag/v8.13.2

The `-all` one.

**Reply by jcupitt** [reply]
I meant to explain that the MRXS file itself is actually a JPEG, so if you use an image viewer which does not support MRXS you just see a JPEG thumbnail.

It sounds like that's what you're seeing, so your libvips is probably missing openslide support. 

Since you're on Windows, you probably downloaded  `vips-dev-w64-web-8.13.2.zip` when you need to download `vips-dev-w64-all-8.13.2.zip`.

There's another complication: 3dhistech have unfortunately changed the MRXS recently and openslide does not support the new version. You will probably not be able to read modern MRXS slides with libvips.
